domain/src/main/java/org/oppia/domain/analytics/LogReportStorageHelper.kt

```diff
@@ -0,0 +1,82 @@
+package org.oppia.domain.analytics
+
+import androidx.lifecycle.LiveData
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import org.oppia.app.model.EventLog
+import org.oppia.app.model.EventLogs
+import org.oppia.data.persistence.PersistentCacheStore
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.data.DataProviders
+import org.oppia.util.logging.Logger
+import org.oppia.util.threading.BackgroundDispatcher
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class LogReportStorageHelper @Inject constructor(
+  private val cacheStoreFactory: PersistentCacheStore.Factory,
+  private val persistentCacheStore: PersistentCacheStore<EventLogs>,
+  private val dataProviders: DataProviders,
+  private val logger: Logger,
+  @BackgroundDispatcher private val backgroundCoroutineDispatcher: CoroutineDispatcher
+) {
+  private val eventLogStore =
+    cacheStoreFactory.create("event_logs", EventLogs.getDefaultInstance())
+  private val coroutineScope = CoroutineScope(backgroundCoroutineDispatcher)
+  private val storeSizeLimit = 10
+
+  fun addEventLog(eventLog: EventLog) {
```

Comment:
As we had discussed, please make this a generic addLog method that works for crash or event logs. Or if that's too difficult, please add addCrashLogs here as well

Reply:
Yes, I tried adding a generic add log method. But I feel having two separate functions for event and exception logging will be better.

Reply:
Discussion done offline.